### PR TITLE
test(e2e): add toast interaction tests

### PR DESCRIPTION
**Pull request by anuraghazra**
## Component Checklist

- [ ] Update Component Status Page
- [ ] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [ ] Add Interaction Tests (if applicable)
- [ ] Add changeset

### CODE REVIEW

File: packages/blade/src/components/Toast/__tests__/Popover.test.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,237 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable import/no-extraneous-dependencies */
```
[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/src/components/Toast/__tests__/Popover.test.stories.tsx:2]: `yarn test:react:interaction`

[the remark this answers is not in the document]
Comment (snitin315) [reply packages/blade/src/components/Toast/__tests__/Popover.test.stories.tsx:2]: I see toast tests are failing. Also, can you point me to the docs with more info like how to write these tests?

Screenshot 2024-02-13 at 1 27 44 PM
Screenshot 2024-02-13 at 1 29 49 PM

[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/src/components/Toast/__tests__/Popover.test.stories.tsx]: Fixed. 

@snitin315 more info on here: https://storybook.js.org/docs/writing-tests/interaction-testing

You can also check example code for Popover, Carousel etc interaction tests.

File: packages/blade/src/components/Toast/__tests__/Toast.test.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+TestToastShow.play = async () => {
+  const { getByRole, queryByText } = within(document.body);
+  await sleep(1000);
+
+  const toastContent = 'Payment successful';
+  await expect(queryByText(toastContent)).not.toBeInTheDocument();
+  const button = getByRole('button', { name: 'Show Toast' });
+  await userEvent.click(button);
+  await sleep(400);
```
Comment (saurabhdaware) [inline packages/blade/src/components/Toast/__tests__/Toast.test.stories.tsx:51]: `await waitFor(() => )` instead of sleep

Comment (anuraghazra) [reply packages/blade/src/components/Toast/__tests__/Toast.test.stories.tsx:51]: Tried it before, doesn't work. We need to discreetly wait for the animations to finish before asserting.

Comment (saurabhdaware) [reply packages/blade/src/components/Toast/__tests__/Toast.test.stories.tsx:51]: The exit would depend on absolute duration no? but I am guessing the toast opens immediately on button click. So can we not use waitFor on those parts? and keep sleep where we want to actually wait for some duration for toast to disappear etc?

Comment (anuraghazra) [reply packages/blade/src/components/Toast/__tests__/Toast.test.stories.tsx:51]: > So can we not use waitFor on those parts

Maybe somethings wrong with storybook testing lib, it doesn't work with waitFor. You need to wait for a discreet amount of time.

**Review by saurabhdaware — APPROVED** [review]

**Pull request merged**